compiler/src/dotty/tools/dotc/transform/ReifyQuotes.scala

```diff
@@ -405,6 +405,14 @@ class ReifyQuotes extends MacroTransform {
             // TODO move to FirstTransform to trigger even without quotes
             cpy.DefDef(tree)(rhs = defaultValue(tree.rhs.tpe))
 
+          case tree: DefTree if level >= 1 =>
+            val newAnnotations = tree.symbol.annotations.mapconserve { annot =>
+              val newAnnotTree = transform(annot.tree) given ctx.withOwner(tree.symbol)
+              if (annot.tree == newAnnotTree) annot
+              else ConcreteAnnotation(newAnnotTree)
+            }
+            tree.symbol.annotations = newAnnotations
+            super.transform(tree)
```

Comment:
Why we need `level > 1`?

Reply:
For code like this
```scala
import scala.quoted._
class Test {
  def foo(str: String) given QuoteContext = '{
   '{
     @deprecated(str, "") // error
      def bar = ???
    }
  }
}
```
I will add a test case

Reply:
Added

Reply:
I'm wondering if the following code is possible and if it's handled by the compiler.

```Scala
import scala.quoted._

class annot[T: Type](x: T)

class Test()(given qtx: QuoteContext) {
  @annot(4)
  def foo(str: String) = ()
}
```